modules/labs-react/combobox/lib/Combobox.tsx

```diff
@@ -184,11 +184,19 @@ const Combobox = ({
   >([]);
   const [announcementText, setAnnouncementText] = useState('');
 
-  // The text input that is wrapped may have a ref already set, if it doesn't we will use our own default.
-  // This default shouldn't be used on it's own, ut we need to declare separately so hooks are always called in the same order.
-  const _defaultInputRef: React.RefObject<HTMLInputElement> = useRef(null);
-  const inputRef: React.RefObject<HTMLInputElement> =
-    (typeof children.props.inputRef !== 'function' && children.props.inputRef) || _defaultInputRef;
+  // Create a ref to the soon-to-be-created TextInput clone for internal use.
+  // Use useForkRef to combine it with the ref already assigned to the original
+  // TextInput (if it exists) to create a single callback ref which can be
+  // forwarded to the TextInput clone.
+  const inputRef: React.RefObject<HTMLInputElement> = useRef(null);
+  // We need access to the original TextInput's ref _property_ (not prop) so we
+  // can combine it with the internal inputRef using useForkRef. ref isn't
+  // listed in the ReactElement interface, but it's there, so we cast children
+  // to satisfy TS.
+  const elementRef = useForkRef(
+    (children as React.ReactElement<TextInputProps> & {ref: React.Ref<HTMLInputElement>}).ref,
+    inputRef
+  );
```

Comment:
Calling out this funky bit of casting I had to do to get TS to stop complaining about the lack of a `ref` property.

@NicholasBoll We discussed this code a bit earlier. I ended up not using `PropsWithAs` in favor of adding `ref` to the casted interface explicitly.